### PR TITLE
Custom Scripts Custom Type Struct Being implemented for npcs 

### PR DESCRIPTION
**Pull request by suday18**
This PR implements the custom type module and specifically the object_t struct to be used within the npc struct. This flexible data structure that can hold either a fundamental data type, such as bool, char, int, etc., or a Lua directory which will be responsible for returning the data. IN our case, we applied this struct to modify the room_id variable which was previously a char*.

Changes:

Updates to the npc.c and npc.h files in the npc directory.
Modifications to the npc_id variable inside these two files and the existing npc struct.

### CODE REVIEW

**Comment by firatciftci** [comment]
@suday18 @tammytran134 @DINO-ADV After handling the merge conflict for you guys, it seems that the code is still failing to pass the check on Travis. I've mentioned this multiple times: Both this PR and the PR of similar nature for npc-rooms _**has never**_ had a successful Travis build. I’ll have to say this is up to you guys to handle, as ultimately, it is your responsibility to write passing code and also tests for your code and PR, not the senior developers. I’ll still try to see why the issue is occurring and if I notice anything I’ll let you know. Please handle the necessary changes, ideally before the deadline of `dev` being frozen.

**Comment by firatciftci** [comment]
Also, I should note that as @tammytran134 is one of the developers assigned to this PR, you guys need to tag someone else who hasn't worked on the PR as the reviewer for the PR.

**Comment by tammytran134** [comment]
@firatciftci Hi Firat, the build test has passed, we are sorry for the inconvenience we've caused.

File: include/npc/npc.h [hunk 1]
```diff
@@ -15,7 +16,7 @@ typedef struct npc {
     UT_hash_handle hh;
 
     /* string identifier of the npc, <21 chars */
```
Comment (firatciftci) [inline include/npc/npc.h:18]: This comment may need to be modified, as `npc_id` is not of `char *` anymore.

File: src/npc/src/npc.c [hunk 2]
```diff
@@ -48,6 +48,7 @@ int npc_free(npc_t *npc)
     {
         convo_free(npc->dialogue);
     }
+    free (npc->npc_id->args);
```
Comment (firatciftci) [inline src/npc/src/npc.c:51]: Minor stylistic thing: There shouldn't be a space after `free` and before `(`.

Comment (tammytran134) [reply src/npc/src/npc.c]: Fixed!

File: include/npc/npc.h (hunk 1 above)
Comment (tammytran134) [reply include/npc/npc.h]: Fixed!

**Comment by firatciftci** [comment]
Issue Score: ✔️+

Comments:
Good work overall! The only thing I'll point out is that, when you create a PR, you are requesting a review of the code in the branch you want to merge. As such, you should not push any commits to the branch right after creating the PR; you should only do so in response to a code review. Make sure that, when you create a PR, the code is ready for review and passes all the tests.

**Pull request merged**